### DISCUSSION TITLE
Disable a certain analyzer

### DISCUSSION THREAD

**Discussion by TimmRuppert** [Ericsson/codechecker#4324] [Q&A] [answered]
Hello, 

I know that I can disable certain checks but can I also disable a complete analyzer? 

In my particular use case I would like to let codechecker run on every new push to an open merge requests. Unfortunately running it takes over 15 minutes on a medium sized project and thus our CI gets clogged. Turing off all clangsa checks helps to reduce the time a lot. 

I know there is a `--disable-all ` flag but I would like to stick to the default proposed cppchecker (and our local .clangtidy file).

**Comment by dkrupp** [comment, marked as answer]
Hi,

you can explicitly enable analyzers you want to be executed:

e.g. 
`CodeChecker analyze ./compile_commands.json --analyzers clang-tidy  cppcheck -o ./report_dir`

Leaving out clangsa will solve your case.

You may also consider running clangsa in shallow mode. This shortens the clang static analyzer execution time considerably (on the cost of missing out some potential bugs). 

`CodeChecker analyze ./compile_commands.json --analyzer-config clangsa:mode=shallow -o ./report_dir`

**Reply by TimmRuppert** [reply]
Thank you very much! This is all I needed and thank again for the bonus tip of the shallow mode option!
